bundle/config/mutator/override_compute.go

```diff
@@ -38,24 +39,37 @@ func overrideJobCompute(j *resources.Job, compute string) {
 }
 
 func (m *overrideCompute) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
-	if b.Config.Bundle.Mode != config.Development {
+	var diags diag.Diagnostics
+
+	if b.Config.Bundle.Mode == config.Production {
 		if b.Config.Bundle.ClusterId != "" {
-			return diag.Errorf("cannot override compute for an target that does not use 'mode: development'")
+			// Overriding compute via a command-line flag for production works, but is not recommended.
+			diags = diags.Extend(diag.Diagnostics{{
+				Summary: "Setting a cluster override for a target that uses 'mode: production' is not recommended",
+				Detail:  "It is recommended to always use the same compute for production target for consistency.",
+			}})
```

Comment:
I tried this out and got:
```
% databricks bundle deploy --target prod --cluster-id ...
Error: Setting a cluster override for a target that uses 'mode: production' is not recommended

It is recommended to always use the same compute for production target for consistency.
```

I believe you intend to warn here?

Reply:
D'oh, yes, fixed.

Reply:
@lennartkats-db I don't see the fix here.

Reply:
https://github.com/databricks/cli/pull/1994